ml-prims/README.md

```diff
@@ -1,44 +1,77 @@
-# ml-prims
-This repo contains most of the ML primitives.
+# Introduction
+This repo contains some of the common infrastructural components as well as
+computational primitives, that will be useful while building a ML algo repo from
+scratch.
 
 # Setup
 ## Dependencies
+1. git
+2. zlib
+3. cmake (>= 3.8)
+4. CUDA SDK (>= 8.0)
 
-1. zlib
-2. cmake (>= 3.8 and <= 3.11.4, version 3.11.4 is recommended and there are some issues with version 3.12)
-3. CUDA SDK (>= 8.0)
-4. Cython (>= 0.28)
-5. gcc (>=5.4.0)
-6. nvcc (this comes with CUDA SDK)
-
-### Building ml-prims:
-
-ml-prims is implemented as header only C++/CUDA libraries for the developers who would like to call these APIs from their projects. You can build and run the Google tests if you are interested in helping us to improve these libraries.
```

Comment:
I believe the `Building ml-prims` heading should go away with #46, though I think the paragraph should stick around, perhaps under a different section (since the build for algos/prims should be consolidated)

Reply:
Done.